pkg/server/retrieve.go

```diff
@@ -68,24 +68,42 @@ func (s *retrieveServlet) retrieve(w http.ResponseWriter, r *http.Request) resul
 		return s.fail(log(ctx, nil).WithField("method", r.Method), w, "method not allowed", "", http.StatusMethodNotAllowed)
 	}
 
-	dateNumber64, err := strconv.ParseUint(vars["day"], 10, 32)
-	if err != nil {
-		return s.fail(log(ctx, err), w, "invalid day parameter", "", http.StatusBadRequest)
-	}
-	dateNumber := uint32(dateNumber64)
+	var startTimestamp time.Time
+	var endTimestamp time.Time
+	var dateNumber uint32
+
+	if config.AppConstants.EnableEntirePeriodBundle == true && vars["day"] == "00000" {
+
+		endDate := timemath.CurrentDateNumber() - 1
+		startDate := endDate - numberOfDaysToServe
+
+		dateNumber = endDate
+
+		startTimestamp = time.Unix(int64(startDate*86400), 0)
+		endTimestamp = time.Unix(int64((endDate+1)*86400), 0)
```

Comment:
Maybe I'm reading this wrong, but it seems like we're doing "yesterday" on L77, then taking numberOfDaysToServe in the past, and then on L83 we're adding back the day we removed so we're back to "today" 
This seems to imply that we're also serving today as part of the bundle, is there ever a use case where we'd want DisableCurrentDateCheckFeatureFlag to be false and EnableEntirePeriodBundle to be true? It seems like we're assuming that EnableEntirePeriodBundle means DisableCurrentDateCheckFeatureFlag is also true.
It's also entirely possible that my timeMath is just bad.

Reply:
endTimestamp is just the outer bound of the period, so you go up to the end days midnight. We need to ensure that yesterdays day is used in `startDate` and `dateNumber` and that the correct endTimestamp is then calculated.